src/account/sync/mod.rs

```diff
@@ -952,7 +929,15 @@ impl SyncedAccount {
         );
 
         let remainder = if has_remainder {
-            selected_addresses.last().cloned()
+            selected_addresses
+                .clone()
```

Comment:
oh i just realized you're cloning the entire list here; this can be improved by cloning the appropriated item instead.